client/extensions/woocommerce/components/product-search/index.js

```diff
@@ -0,0 +1,81 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { trim } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	fetchProductSearchResults,
+	clearProductSearch,
+} from 'woocommerce/state/sites/products/actions';
+import ProductSearchResults from './results';
+import SearchCard from 'components/search-card';
+
+class ProductSearch extends Component {
+	static propTypes = {
+		onSelect: PropTypes.func.isRequired,
+	};
+
+	state = {
+		query: '',
+	};
+
+	handleSearch = query => {
+		const { siteId } = this.props;
```

Comment:
Optional. Consider debouncing this. As the number of products on a site grows, I suspect these search queries could take some time/resources to complete

Reply:
Nevermind. TIL about delaySearch :P